doc/configure

```diff
@@ -72,43 +82,64 @@ RULES = [{
 }]
 
 
+def make_html_unique(man, html_man, html_mans, builds):
+    html_unique = HTML_UNIQUE_DIR / f"{man.stem}.html"
+    builds.append({
+        "inputs": [html_man, BIN_DIR / "uniquify-header-ids"],
+        "outputs": [html_unique],
+        "rule": "uniquify_header_ids",
+        "variables": {
+            "man-name": man.stem,
+            "in-file": html_man,
+        }
+    })
+    html_mans.append(html_unique)
+
+
+def roff_to_html(man, roff_out, builds):
+    html_man = HTML_MAN_SRC_DIR / f"{man.stem}.html"
+    builds.append({
+        "inputs": [roff_out],
+        "outputs": [html_man],
+        "rule": "roff_to_html",
+        "variables": {
+            "man-name": man.stem,
+        }
+    })
+    return html_man
+
+
 def main():
     builds = []
-
     html_mans = []
-    for man in (SRC_DIR / "man").iterdir():
+
+    for man in (SRC_DIR / "voluptuous-man").iterdir():
         roff_out = ROFF_DIR / f"{man.stem}.roff"
         builds.append({
             "inputs": [man],
             "outputs": [roff_out],
-            "rule": "sc_to_roff",
-            "variables": {
-                "man-name": man.stem,
-            }
-        })
-
-        html_man = HTML_MAN_SRC_DIR / f"{man.stem}.html"
-        builds.append({
-            "inputs": [roff_out],
-            "outputs": [html_man],
-            "rule": "roff_to_html",
+            "rule": "voluptuous_to_roff",
             "variables": {
                 "man-name": man.stem,
+                "data-path": man.resolve(),
             }
         })
+        html_man = roff_to_html(man, roff_out, builds)
+        make_html_unique(man, html_man, html_mans, builds)
 
-        html_unique = HTML_UNIQUE_DIR / f"{man.stem}.html"
+    for man in (SRC_DIR / "man").iterdir():
+        roff_out = ROFF_DIR / f"{man.stem}.roff"
         builds.append({
-            "inputs": [html_man, BIN_DIR / "uniquify-header-ids"],
-            "outputs": [html_unique],
-            "rule": "uniquify_header_ids",
+            "inputs": [man],
+            "outputs": [roff_out],
+            "rule": "sc_to_roff",
             "variables": {
                 "man-name": man.stem,
```

Comment:
What about creating a new function such as the one below and calling it twice within `main()`?

```python
def convert_to_roff(builds, html_mans, man_dir, rule, has_data_path=False):
    for man in (SRC_DIR / man_dir).iterdir():
        roff_out = ROFF_DIR / f"{man.stem}.roff"
        builds.append({
                "inputs": [man],
                "outputs": [roff_out],
                "rule": rule,
                "variables": {
                    "man-name": man.stem
                }
        })
        if has_data_path:
            builds["variables"]["data-path"] = man.resolve()
        html_man = roff_to_html(man, roff_out, builds)
        make_html_unique(man, html_man, html_mans, builds)
    return [builds, html_mans]


def main():
    builds = []
    html_mans = []
    builds, html_mans = convert_to_roff(
        builds,
        html_mans,
        "voluptuous-man",
        "voluptuous_to_roff",
        has_data_path=True
    )
    builds, html_mans = convert_to_roff(
        builds,
        html_mans,
        "man",
        "sc_to_roff"
    )
    builds.append({
        "inputs": html_mans + [
            BIN_DIR / "build-html-doc",
            TEMPLATE_DIR / "index.jinja.html",
        ],
        "outputs": [DOC_DIR / "out" / "html"/ "index.html"],
        "rule": "build_html_doc",
        "variables": {
            "html-mans": " ".join([str(h) for h in html_mans]),
        }
    })
    for build in builds:
        for k, v in build.items():
            if isinstance(v, list):
                build[k] = [str(s) for s in v]
        try:
            build["implicit"].append(str(DOC_DIR / "configure"))
        except KeyError:
            build["implicit"] = [str(DOC_DIR / "configure")]
    with open("build.ninja", "w") as handle:
        ninja = lib.ninja_syntax.Writer(handle)
        for rule in RULES:
            ninja.rule(**rule)
        for build in builds:
            ninja.build(**build)
```

Reply:
Done in 8fa8c06